src/viam/resource/manager.py

```diff
@@ -105,3 +105,7 @@ def _resource_by_name_only(self, name: str) -> ResourceBase:
             if rname.name == name:
                 return resource
         raise ResourceNotFoundError("resource", name)
+
+    async def close(self):
+        for _, resource in self.resources.items():
+            await resource.close()
```

Comment:
what if a resource errors while closing?

Reply:
Ah, I guess the move here is to raise an error, but continue with the close so that the robot isn't just sitting there not able to close.

Reply:
Maybe you could collect the errors (or at least which resources did error) and then raise/log those?

pseudocode below

```python
for _, resource in self.resource.items():
    try:
        await resource.close()
    catch e:
        LOGGER.error(f'Encountered error while closing resource named {resource.name}', e)
```

or even

```python
errors = {}
for _, resource in self.resource.items():
    try:
        await resource.close()
    catch e:
        errors[resource.name] = e
    finally:
        LOGGER.error(f'Encountered {len(errors)} errors while closing: {errors}')
```

Reply:
Yep, I did something similar to the second one!